src/Maui/Prism.Maui/Navigation/Xaml/Navigation.cs

```diff
@@ -134,7 +134,7 @@ public static IContainerProvider GetContainerProvider(this BindableObject bindab
             return container;
         else if(bindable is Page page)
         {
-            if (page.Parent is FlyoutPage flyout && flyout.Flyout == page)
+            if (page.Parent is FlyoutPage flyout && flyout.Detail == page)
```

Comment:
I'm not sure you understand what this code is doing. This doesn't exactly affect Navigation. It pulls the Scoped Container Provider from the FlyoutPage when the current page is the FlyoutPage's Flyout since these are conceptually a single page for Prism.